mlrun/track/trackers/mlflow_tracker.py

```diff
@@ -0,0 +1,449 @@
+        with zipfile.ZipFile(archive_path, "w") as zip_file:
+            for path in model_uri.rglob("*"):
+                zip_file.write(filename=path, arcname=path.relative_to(model_uri))
+
+        # Get inputs and outputs info:
+        inputs = outputs = None
+        if model_info.signature is not None:
+            if model_info.signature.inputs is not None:
+                inputs = MLFlowTracker._schema_to_feature(
+                    schema=model_info.signature.inputs
+                )
+            if model_info.signature.outputs is not None:
+                outputs = MLFlowTracker._schema_to_feature(
+                    schema=model_info.signature.outputs
+                )
+
+        # Log the model:
+        return context_or_project.log_model(
+            key=key,
+            framework="mlflow",
+            model_file=str(archive_path),
+            metrics=metrics,
+            labels={
+                "mlflow_run_id": model_info.run_id,
+                "mlflow_version": model_info.mlflow_version,
+                "mlflow_model_uuid": model_info.model_uuid,
+            },
+            extra_data=extra_data,
+            inputs=inputs,
+            outputs=outputs,
+        )
+
+    @staticmethod
+    def _log_artifact(
+        context_or_project: Union[MLClientCtx, MlrunProject],
+        key: str,
+        local_path: str,
+        tmp_path: str,
+    ) -> Artifact:
+        """
+        Log the given produced file from MLFlow as a run artifact in MLRun.
+
+        :param context_or_project: The MLRun context or project to log to.
+        :param key:                The artifact's key.
+        :param local_path:         The local path to the artifact.
+        :param tmp_path:           The path to the dir where we temporarily save artifacts
+        """
+        # Check if the artifact is a directory for archiving it:
+        if pathlib.Path(local_path).is_dir():
+            # Prepare the archive path:
+            archive_path = pathlib.Path(tmp_path) / f"{key}.zip"
+            local_path = pathlib.Path(local_path)
+            # Zip the artifact:
+            with zipfile.ZipFile(archive_path, "w") as zip_file:
+                for path in local_path.rglob("*"):
+                    zip_file.write(filename=path, arcname=path.relative_to(local_path))
+            # Set the local path to the archive file:
+            local_path = str(archive_path)
+
+        # Log and return the artifact in the local path:
+        return context_or_project.log_artifact(
+            item=key,
+            local_path=local_path,
+        )
+
+    @staticmethod
+    def _schema_to_feature(schema: mlflow.types.Schema) -> List[Feature]:
+        """
+        Cast MLFlow schema to MLRun features.
+
+        :param schema: The MLFlow schema.
+
+        :return: List of MLRun features representing the schema.
+        """
+        # Import here due to circular dependencies:
+        from mlrun.frameworks._common import CommonUtils
```

Comment:
Generally, I would ask you to solve the import loop. Since its a big PR I'm fine with it for now but keep in mind that import loops are a symptom for bad design and I believe we can solve them